iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2247,7 +2249,19 @@ static int SendMqttConnectMsg(PMQTTTRANSPORT_HANDLE_DATA transport_data)
                 }
                 else
                 {
-                    transport_data->isProductInfoSet = true;
+                    STRING_HANDLE param;
+                    const char* dt_model_id = transport_data->transport_callbacks.dt_model_id_cb(transport_data->transport_ctx);
+                    if (dt_model_id != NULL)
+                    {
+                        param = STRING_construct_sprintf("&%s=%s", DT_MODEL_ID_TOKEN, STRING_c_str(URL_EncodeString(dt_model_id)));
+
+                        if (STRING_concat_with_STRING(transport_data->configPassedThroughUsername, param) != 0)
+                        {
+                            LogError("Failed to set DT Model Id");
+                        }
+                    }
+
+                    transport_data->isOptionalParameterSet = true;
```

Comment:
should this be set even after a LogError failure?

Reply:
the optional parameters are only supposed to be set once 

---
In reply to: [385312375](https://github.com/Azure/azure-iot-sdk-c/pull/1439#discussion_r385312375) [](ancestors = 385312375)

Reply:
sure but even if it fails? I'm ignorant on this so not sure. Just seems odd to have it fail but then say `true` that it was set.